UnitsNet/CustomCode/UnitAbbreviationsCache.cs

```diff
@@ -395,77 +419,33 @@ private IReadOnlyList<string> ReadAbbreviationsFromResourceFile(string? quantity
-            /// </summary>
-            public string UnitTypeName { get; }
-
-            /// <summary>
-            ///     The unit name, such as "Centimeter".
-            /// </summary>
-            public string UnitName { get; }
-
-            /// <summary>
-            ///     The culture name, such as "en-US".
-            /// </summary>
-            public string CultureName { get; }
-
-            [SuppressMessage("ReSharper", "InconsistentNaming", Justification = "Matches record naming.")]
-            public AbbreviationMapKey(string UnitTypeName, string UnitName, string CultureName)
-            {
-                this.UnitTypeName = UnitTypeName;
-                this.UnitName = UnitName;
-                this.CultureName = CultureName;
-            }
-
-            public bool Equals(AbbreviationMapKey? other)
-            {
-                if (ReferenceEquals(null, other)) return false;
-                if (ReferenceEquals(this, other)) return true;
-                return UnitTypeName == other.UnitTypeName && UnitName == other.UnitName && CultureName == other.CultureName;
-            }
-
-            public override bool Equals(object? obj)
-            {
-                if (ReferenceEquals(null, obj)) return false;
-                if (ReferenceEquals(this, obj)) return true;
-                if (obj.GetType() != GetType()) return false;
-                return Equals((AbbreviationMapKey)obj);
-            }
-
-            public override int GetHashCode()
-            {
-                unchecked
-                {
-                    int hashCode = UnitTypeName.GetHashCode();
-                    hashCode = (hashCode * 397) ^ UnitName.GetHashCode();
-                    hashCode = (hashCode * 397) ^ CultureName.GetHashCode();
-                    return hashCode;
-                }
-            }
+            // TODO this is certain to have terrible performance (especially on the first run)
+            // TODO we should consider adding a (lazy) dictionary for these
+            // Use case-sensitive match to reduce ambiguity.
+            // Don't use UnitParser.TryParse() here, since it allows case-insensitive match per quantity as long as there are no ambiguous abbreviations for
+            // units of that quantity, but here we try all quantities and this results in too high of a chance for ambiguous matches,
+            // such as "cm" matching both LengthUnit.Centimeter (cm) and MolarityUnit.CentimolePerLiter (cM).
+            return QuantityInfoLookup.Infos
+                .SelectMany(quantityInfo => quantityInfo.UnitInfos)
+                .Select(unitInfo => GetAbbreviations(unitInfo, formatProvider).Contains(unitAbbreviation, StringComparer.Ordinal)
+                    ? unitInfo
+                    : null)
+                .Where(unitValue => unitValue != null)
+                .Select(unitValue => unitValue!)
+                .ToList();
```

Comment:
Actually, this has no business being here- any `abbreviation matching` methods should be part of the `UnitParser`.